### ISSUE TITLE
Replace the mod ReAuth with OAuth

### ISSUE THREAD

**Issue by Emtech2** [Divine-Journey-2/main#252]
**Modpack version:**
2.2.0

**Describe the request:**
Replace the mod ReAuth with OAuth

**Reasons why this should be added:**
ReAuth does not currently support Microsoft accounts and does not appear to be adding support. OAuth has added Microsoft Account support while maintaining Mojang account support and is open source. 

**Describe alternatives you've considered:**
The only alternative is any user who has a Microsoft account login manually adding OAuth and removing ReAuth.

**Additional context:**
Launcher assets have stated deadlines for migrating will be early 2022. However, any new accounts are already made as Microsoft accounts.
The CurseForge link: https://www.curseforge.com/minecraft/mc-mods/oauth
Source: https://github.com/Sintinium/oauth

**Comment by thecodingchicken** [comment]
In fact, I just came across the same issue.  I will be giving it a test over the next few days.  Until then, I don't think we should have it put into the main branch of the pack.

**Comment by Atricos** [comment]
The 1.12.2 versions have 16 downloads in total.
![image](https://user-images.githubusercontent.com/24854267/132650298-4162bb86-5c67-467b-9152-3ebec273fbdf.png)

I would like to wait before all major bugs are found and fixed, before I add it to the pack. Until then, feel free to replace ReAuth inside mods and add OAuth to your pack. It's a client-side mod, so there shouldn't be any issues.

**Comment by thecodingchicken** [comment]
Agreed.  I don't have much time to play right now, but I will test it the whole auth stuff out.

**Comment by Sintinium** [comment]
I just saw this issue and decided to backport all the fixes and features from 1.16/1.17 which has no issues so far with ~3,000 downloads. If you do consider adding it and find any bugs let me know and I'll get on it asap.
